qbft/spectest/tests/commit/decided_full_data.go

```diff
@@ -1,8 +1,47 @@
 package commit
 
-import "github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+	"github.com/herumi/bls-eth-go-binary/bls"
+)
 
 // DecidedFullData tests the aggregated commit returned when decided to include the full data
 func DecidedFullData() tests.SpecTest {
-	panic("implement")
+	ks := testingutils.Testing4SharesSet()
+
+	msgs := []*qbft.SignedMessage{
+		testingutils.TestingProposalMessage(ks.Shares[1], 1),
+
+		testingutils.TestingPrepareMessage(ks.Shares[1], 1),
+		testingutils.TestingPrepareMessage(ks.Shares[2], 2),
+		testingutils.TestingPrepareMessage(ks.Shares[3], 3),
+
+		testingutils.TestingCommitMessage(ks.Shares[1], 1),
+		testingutils.TestingCommitMessage(ks.Shares[2], 2),
+		testingutils.TestingCommitMessage(ks.Shares[3], 3),
+	}
+
+	outMsgs := []*qbft.SignedMessage{
+		testingutils.TestingPrepareMessage(ks.Shares[1], 1),
+		testingutils.TestingCommitMessage(ks.Shares[1], 1),
+		testingutils.TestingCommitMultiSignerMessage([]*bls.SecretKey{ks.Shares[1], ks.Shares[2], ks.Shares[3]}, []types.OperatorID{1, 2, 3}),
+	}
+
+	return &tests.ControllerSpecTest{
```

Comment:
I used the `ControllerSpecTest` because the I needed the `controller` structure since it is the one that broadcasts the decided messages.

Reply:
What should be the correct comment / instruction for this test?

Reply:
I think that the comment should simply be moved to `controller/valid_full_data.go` and here it can be deleted imho.
I don't see much difference between the 2.

If you can think of a way to make a meaningful `MsgProcessingTest` for this be my guest... but I am not sure how

Reply:
Yup, agree 100%